x-pack/plugins/ml/public/application/components/ml_inference/components/reindex_with_pipeline.tsx

```diff
@@ -0,0 +1,407 @@
+      setDestinationIndexInvalidMessage(destIndexEmpty);
+    } else if (!isValidIndexName(e.target.value)) {
+      setDestinationIndexInvalidMessage(destIndexInvalid);
+    } else {
+      setDestinationIndexInvalidMessage(undefined);
+    }
+  };
+
+  useEffect(
+    function canReindexCheck() {
+      async function checkPrivileges() {
+        try {
+          const privilege = await hasPrivileges({
+            index: [
+              {
+                names: [selectedIndex[0].label], // uses wildcard
+                privileges: ['read'],
+              },
+              {
+                names: [destinationIndex], // uses wildcard
+                privileges: ['write'],
+              },
+            ],
+          });
+          if (!privilege) {
+            setCanReindex(false);
+          }
+          setCanReindex(
+            privilege.securityDisabled === true || privilege.has_all_requested === true
+          );
+        } catch (e) {
+          const error = extractErrorMessage(e);
+          const errorMessage = i18n.translate(
+            'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.review.indexPrivilegeErrorMessage',
+            {
+              defaultMessage: 'User does not have required permissions to reindex {index}. {error}',
+              values: { error, index: selectedIndex[0].label },
+            }
+          );
+          setCanReindexError(errorMessage);
+        }
+      }
+      if (hasPrivileges !== undefined) {
+        checkPrivileges();
+      }
+    },
+    [hasPrivileges, sourceIndex, toasts, destinationIndex, selectedIndex]
+  );
+
+  useEffect(
+    function checkDestIndexExists() {
+      if (destinationIndex !== undefined && destinationIndex !== '') {
+        debouncedIndexCheck();
+      }
+    },
+    [destinationIndex, debouncedIndexCheck]
+  );
+
+  useEffect(
+    function getIndexOptions() {
+      async function getAllIndices() {
+        const indices = await getIndices();
+        const indexOptions = indices.map((index) => ({ label: index.name }));
+        setOptions(indexOptions);
+      }
+      getAllIndices();
+    },
+    [getIndices]
+  );
+
+  useEffect(
+    function createDiscoverLink() {
+      async function createDataView() {
+        const dataViewCreationResult = await createKibanaDataView(destinationIndex, data.dataViews);
+        if (
+          dataViewCreationResult?.success === true &&
+          dataViewCreationResult?.dataViewId &&
+          showDiscoverLink
+        ) {
+          const url = await generateDiscoverUrl(dataViewCreationResult.dataViewId);
+          setDiscoverLink(url);
+        }
+      }
+      if (reindexingTaskId !== undefined && shouldCreateDataView === true) {
+        createDataView();
+      }
+    },
+    [
+      reindexingTaskId,
+      destinationIndex,
+      data?.dataViews,
+      generateDiscoverUrl,
+      showDiscoverLink,
+      shouldCreateDataView,
+    ]
+  );
+
+  const reindexButton = (
```

Comment:
Can this button be disabled if the destination index name already exists?

<img width="418" alt="image" src="https://github.com/elastic/kibana/assets/7405507/766b0b5d-003f-48ce-ad6c-81983ca630b9">

Reply:
Updated in 8a13aecbdc39a338a76e70fc8101392fe3bc5eda